misc/plugutils.c

```diff
@@ -37,6 +37,51 @@
 /* Used to allow debugging for openvas-nasl */
 int global_nasl_debug = 0;
 
+/* Track amount of KB memory used */
+static size_t kb_usage = 0;
+
+static size_t max_kb_usage;
+
+void
+init_kb_usage (void)
+{
+  const char *usage_char;
+  int usage_int;
+  usage_char = prefs_get ("max_mem_kb");
+  if (usage_char)
+    {
+      usage_int = atoi (usage_char);
+      if (usage_int < 0)
+        {
+          max_kb_usage = 0;
+        }
+      else
+        {
+          max_kb_usage = (size_t) usage_int * 1024 * 1024;
+        }
+    }
+  else
+    max_kb_usage = 0;
+}
+
+static int
+add_kb_usage (struct script_infos *args, size_t size)
+{
+  if (max_kb_usage == 0)
+    return 0;
+  if (kb_usage > max_kb_usage)
+    return -1;
+  kb_usage += size;
+  if (kb_usage > max_kb_usage)
+    {
+      g_warning ("KB usage exceeded %lu MB. Unable to store any further KB "
```

Comment:
I would add a flag to show this message once per script per host. Otherwise it will flood the log file.

Reply:
It is only shown once per script per host, as it is only shown the first time the limit is reached. After that the function returns immediately